powershell/internal/Set-MaesterAppPermissions.ps1

```diff
@@ -0,0 +1,92 @@
+﻿<#
+    .SYNOPSIS
+    Internal helper function to set Graph API permissions for a Maester application.
+
+    .DESCRIPTION
+    This function configures the required Graph API permissions (app roles) for a Maester application
+    in Azure AD/Entra ID. It handles both creating new permissions and updating existing ones.
+
+    .PARAMETER AppId
+    The Application (Client) ID of the app to configure permissions for.
+
+    .PARAMETER Scopes
+    Array of Graph API permission scopes to configure for the application.
+
+    .EXAMPLE
+    Set-MaesterAppPermission -AppId "12345678-1234-1234-1234-123456789012" -Scopes @("Directory.Read.All", "Policy.Read.All")
+#>
+function Set-MaesterAppPermission {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingWriteHost', '', Justification = 'Colors are beautiful')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseShouldProcessForStateChangingFunctions', '', Justification = 'TODO: Implement ShouldProcess')]
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $AppId,
+
+        [Parameter(Mandatory = $true)]
+        [string[]] $Scopes
+    )
+
+    process {
+        Write-Verbose "Setting permissions for app: $AppId"
+        Write-Verbose "Scopes: $($Scopes -join ', ')"
+
+        # Get the service principal for the app
+        $appSPResponse = Invoke-MtAzureRequest -RelativeUri "servicePrincipals" -Filter "appId eq '$AppId'" -Method GET -Graph
+        $appSPs = $appSPResponse.value
+
+        if ($appSPs.Count -eq 0) {
+            throw "Service principal for application not found. Ensure the application has a service principal."
```

Comment:
If the app ID is not found, this `throw` will end the function immediately. It might be helpful to still allow the Graph application to get checked before ending so admins can see if that SP might be usable.